### PR TITLE
Linter, workspace & docs

### PR DESCRIPTION
**Pull request by szmktk**
Add a Github Action workflow definition with a default config of `golangci-lint`.

[`golangci-lint`](https://golangci-lint.run/) is a linter aggregator for Go projects.
Linters enabled by default https://golangci-lint.run/usage/linters/#enabled-by-default

Also adds `go.work` and useful information to `README.md`

[Check this out](https://github.com/hs3city/goexplorers/actions/runs/7601022396/job/20699934780), all green 🟢

### CODE REVIEW

**Comment by DoomHammer** [comment]
Maybe also https://pre-commit.com/ so most problems could be caught before pushing?

**Comment by szmktk** [comment]
@DoomHammer Of course I considered it but [did not find](https://pre-commit.com/hooks.html) and any sets of hooks that would be actively maintained and as widely adopted by the Go community as `golangci` is.

If you insisted on having a pre-commit hook I'd then suggest to invoke `golangci` from `.git/hooks/pre-commit` script directly.

File: cyoa/main/main.go [hunk 1]
```diff
@@ -2,54 +2,62 @@ package main
...
 	"net/http"
 	"os"
 	"strings"
 )
 
 func main() {
-	fmt.Println("Server starting...")
+	log.Println("Server starting...")
 	http.HandleFunc("/", handlerHttp)
-	http.ListenAndServe(":8080", nil)
```
Comment (szmktk) [inline cyoa/main/main.go:15]: This is what `errcheck` linter has to say about this line:

> Error return value of `http.ListenAndServe` is not checked (errcheck)

File: urlshort/main/main.go [hunk 2]
```diff
@@ -23,32 +23,20 @@ func main() {
...
-	jsonHandler, err := urlshort.JSONHandler([]byte(json), yamlHandler)
-	if err != nil {
-		panic(err)
+	logger.Info("Starting the server on :8080")
+	if err := http.ListenAndServe(":8080", jsonHandler); err != nil {
+		logger.Error("Error starting server", err)
 	}
 
-	logger.Info("Starting the server on :8080")
-	http.ListenAndServe(":8080", jsonHandler)
```
Comment (szmktk) [inline urlshort/main/main.go:51]: This is what `errcheck` linter has to say about this line:

> Error return value of `http.ListenAndServe` is not checked (errcheck)

File: urlshort/main/main.go [hunk 3]
```diff
@@ -60,3 +48,11 @@ func defaultMux() *http.ServeMux {
 func hello(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintln(w, "Hello, world!")
 }
+
+func catch[T any](val T, err error) T {
```
Comment (miniMax18) [inline urlshort/main/main.go:52]: 👍 This "last" generics knowledge is nice to see here 👍

**Pull request merged**